docs/design/resources.md

```diff
@@ -0,0 +1,96 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but may also involve
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding it's bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname,
+liveness status, the peer's current generation, and if it's currently
+the leader of the etcd cluster.
+
+When reconciling the etcd peer resource the operator will do the
+following:
+
+1. If storage is enabled and a correctly named persistent volume claim
+   (PVC) does not exist, create one with the settings provided on the
+   peer resource.
```

Comment:
Are we going to allow people to 'disable' storage? I guess this would be defined as using an emptyDir instead of a PVC?

It'll be cool to allow this model, even if simply to make it easy to run disaster recovery tests (i.e. keep deleting pods and ensure that quorum is maintained 😄)

Reply:
I figured if you omit the storage section of the `EtcdCluster` configuration it just wouldn't configure storage?

Reply:
I think this may be non-trivial as when the Pod is deleted, we'd need to make sure to de-register & re-register the member from the cluster else it will not be able to start properly, from what I understand. But I could be wrong 😄

Reply:
I'd assumed, and this is an assumption, that if a node rejoined with a blank data directory it'd just resync data again. Maybe requires some experimentation. If that's not the case how could an etcd cluster ever recover from single node failure without manual intervention?

Do you want me to remove the 'optional storage' bit for now?

Reply:
It seems simpler given we aren't sure - this serves as a nice document of the initial goals we are working towards, and we can take bullet points from this to form tickets 😄

Let's not aim to support ephemeral storage *at least initially*, until there's a need for it (i.e. for testing disaster scenarios easily, etc)

Reply:
Okay. Updated.